contracts/Rewards/RewardsDistributor.sol

```diff
@@ -18,15 +19,21 @@ import { MaxLoopsLimitHelper } from "../MaxLoopsLimitHelper.sol";
  * Users can receive additional rewards through a `RewardsDistributor`. Each `RewardsDistributor` proxy is initialized with a specific reward
  * token and `Comptroller`, which can then distribute the reward token to users that supply or borrow in the associated pool.
  * Authorized users can set the reward token borrow and supply speeds for each market in the pool. This sets a fixed amount of reward
- * token to be released each block for borrowers and suppliers, which is distributed based on a user’s percentage of the borrows or supplies
+ * token to be released each slot (block or second) for borrowers and suppliers, which is distributed based on a user’s percentage of the borrows or supplies
  * respectively. The owner can also set up reward distributions to contributor addresses (distinct from suppliers and borrowers) by setting
- * their contributor reward token speed, which similarly allocates a fixed amount of reward token per block.
+ * their contributor reward token speed, which similarly allocates a fixed amount of reward token per slot (block or second).
  *
  * The owner has the ability to transfer any amount of reward tokens held by the contract to any other address. Rewards are not distributed
  * automatically and must be claimed by a user calling `claimRewardToken()`. Users should be aware that it is up to the owner and other centralized
  * entities to ensure that the `RewardsDistributor` holds enough tokens to distribute the accumulated rewards of users and contributors.
  */
-contract RewardsDistributor is ExponentialNoError, Ownable2StepUpgradeable, AccessControlledV8, MaxLoopsLimitHelper {
+contract RewardsDistributor is
+    TimeManagerV8,
```

Comment:
RewardsDistributor itself defines storage variables too, so due to gap variable there will be storage collision I think.
We will have to remove gap variable to avoid it.

Reply:
Or we can make a storage contract for the RewardsDistributor states which would be inherited in the RewardsDistributor contract to prevent storage collision and then inherit the TimeManager, in this way there is no need to remove the gap from TimeManager.

Reply:
I would go with the second option. We did something similar in the past for other contracts

Reply:
Done https://github.com/VenusProtocol/isolated-pools/pull/324/commits/d6311f25cf84ac1c82a143109e45ad9fea1d4dc3
These are the snippets of storage verifying that storage layout is correct:
![Screenshot from 2024-01-12 15-04-19](https://github.com/VenusProtocol/isolated-pools/assets/104391977/2f6cf3cd-4e0e-40d8-8b14-3b09370f7871)
![Screenshot from 2024-01-12 15-03-12](https://github.com/VenusProtocol/isolated-pools/assets/104391977/4a1a4624-5b6d-41d1-b9e1-c233402323ea)